--- bufexplorer/plugin/bufexplorer.vim	2010-11-08 13:54:18.000000000 -0600
+++ bufexplorer/plugin/bufexplorer.vim.new	2012-01-29 17:35:29.226148586 -0600
@@ -183,7 +183,11 @@
   elseif empty(l) || index(l, b) == -1
     " Add new buffer to this tab buffer list
     let l = add(l, b)
-    let s:tabSpace[tabpagenr()] = l
+    " let s:tabSpace[tabpagenr()] = l
+
+    " note: it will be out of range
+    " original solution: let s:tabSpace[tabpagenr()] = l
+    call add(s:tabSpace, l )
 
     if g:bufExplorerOnlyOneTab == 1
       " If a buffer can only be available in 1 tab page
